scripts/merge_lora.py

```diff
@@ -70,6 +81,9 @@ def merge_lora(
 
     merge_lora_weights(model)
 
+    if quantize:
```

Comment:
Just realized that it doesn't make sense.
The whole idea is that if someone used quantization during fine-tuning, it's because there was not enough space in VRAM to fit the whole model in other precisions.
That's why merging weights while the model is quantized is a good idea.

But, and there is a big but, if the model is quantized, then the weights will be saved to disk in a quantizaed form too. And the biggest problem here is that the shape of weight matrices will be different to a non-quantized model (each torch.uint8 data type contains 2x nf4 data entries). That can lead to problems down the line.

The obvious idea would be to dequantize the model first and then save, but here we can easily get OOM.

Ideally we should iterate over layers, dequantize one layer at a time and incrementally save the weights on a disk.

Or to support loading a quantized model which functionality is already [added](https://github.com/TimDettmers/bitsandbytes/pull/753) in BNB.

Reply:
How big is memory a concern here? The merging would happen when we are not training, so this should be fine no?

Reply:
@Andrei-Aksionov during fine-tuning you have weights, activations, gradients, optimizer state. If you just need to store the weights, the 7B model will fit in 24GB with bfloat16 precision.

Reply:
@awaelchli Good point: if the training with all the activations and optimizer states could fit in the model, then we have a high chance that the weights alone, even in dequantized form, should fit too, but it's not guaranteed. And I didn't measure the impact, just literally thought about it during my morning tea routine 😄.

But the problem with the script is that, if quantize argument is provided, we first quantize model, merge weights and, before saving to disk, dequantize it again. The memory saving will be only during merging and I don't think that these saving are significant.
We can keep it as is for now, but need to revisit it in the future.

Reply:
Oh I see what you're saying. Yeah probably we can just skip this script and keep the rest.

Reply:
Will do.